src/github.com/getlantern/flashlight/logging/logging_test.go

```diff
@@ -23,12 +23,17 @@ func TestUserAgent(t *testing.T) {
 	listener := func(a string) {
 		userAgent.Set(a)
 	}
-	AddUserAgentListener(listener)
-	RegisterUserAgent(agent)
 
-	// Just register again to hit the code path for duplicates.
+	// Do an initial register just to test the duplicate agent paths.
 	RegisterUserAgent(agent)
 
+	AddUserAgentListener(listener)
+
+	go func() {
+		time.Sleep(200 * time.Millisecond)
+		RegisterUserAgent(agent)
+	}()
```

Comment:
Here's the change to test that setting the user agent asynchronously works fine @uaalto